jupyterlab_gcedetails/src/components/hardware_scaling_status.spec.tsx

```diff
@@ -0,0 +1,218 @@
+        .find(Button)
+        .first()
+        .simulate('click');
+      expect(mockOnDialogClose).toBeCalled();
+    }, 1000);
+  });
+
+  it('Renders with auth error', async () => {
+    mockAuthTokenRetrieval.mockImplementation(() => {
+      throw new Error();
+    });
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    expect(hardwareScalingStatus.state('status')).toEqual(Status.Error);
+    expect(hardwareScalingStatus).toMatchSnapshot('Auth Error');
+  });
+
+  it('Renders with stop error', async () => {
+    mockAuthTokenRetrieval.mockResolvedValue(mockToken);
+    mockStopInstance.mockImplementation(() => {
+      throw new Error();
+    });
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    await mockAuthTokenRetrieval();
+    expect(hardwareScalingStatus.state('status')).toEqual(Status.Error);
+    expect(hardwareScalingStatus).toMatchSnapshot('Stop instance error');
+  });
+
+  it('Renders with operation error', async () => {
+    const mockGetUtilizationData = jest.fn();
+    const mockStopInstance = jest.fn();
+    mockSetMachineType.mockImplementation(() => {
+      throw new Error();
+    });
+    mockSetAccelerator.mockImplementation(() => {
+      throw new Error();
+    });
+    mockStartInstance.mockResolvedValue(mockInstance);
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const resolveValue = Promise.resolve({ ...detailsResponse });
+    mockGetUtilizationData.mockReturnValue(resolveValue);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    await mockAuthTokenRetrieval();
+    await mockStopInstance();
+    await mockStartInstance();
+    await mockGetUtilizationData();
+    setTimeout(() => {
```

Comment:
not sure if there is a better way to do this but it would register the expect too fast before the state actually changed, hence the timeout

Reply:
Generally you want to avoid setting timeouts in your tests because they slow them down considerably. Let's talk about this one in our standup.